sql/src/main/java/org/apache/druid/sql/http/SqlResource.java

```diff
@@ -117,171 +108,41 @@
   @POST
   @Produces(MediaType.APPLICATION_JSON)
   @Consumes(MediaType.APPLICATION_JSON)
+  @Nullable
   public Response doPost(
       final SqlQuery sqlQuery,
       @Context final HttpServletRequest req
-  ) throws IOException
+  )
```

Comment:
Thank you for pulling the response out of the body of this method. That's been something that's been bugging me for a while.

This refactoring will be helpful to the gRPC API project: the HTTP request handler is now just a few lines of code which is easy to replicate for the gRPC handler.